src/components/molecules/SecretPasswordForm/SecretPasswordForm.jsx

```diff
@@ -23,22 +23,28 @@ const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
     setError(false);
   }
 
-  function passwordSubmitted(e) {
-    e.preventDefault();
-    setMessage("Checking password...");
+  const passwordSubmitted = useCallback(
+    async (e) => {
+      e.preventDefault();
+      setMessage("Checking password...");
+
+      try {
+        const result = await checkAccess({
+          venueId,
+          password,
+        });
+        setLocalStorageToken(venueId, result.data.token);
 
-    const checkPassword = firebase.functions().httpsCallable("checkPassword");
-    checkPassword({ venue: venueId, password: password })
-      .then(() => {
         setInvalidPassword(false);
         setMessage("Password OK! Proceeding...");
         history.push(venueEntranceUrl(venueId));
-      })
-      .catch(() => {
+      } catch (error) {
         setInvalidPassword(true);
-        setMessage(null);
-      });
-  }
+        setMessage(`Password error: ${error.toString()}`);
+      }
```

Comment:
This was probably cleaner/more succinct using the `.then()`/`.catch()` functions on the `Promise`, rather than using `await` and then having to wrap it in a `try`/`catch`. Then we could have just `await`'d on the 'main `Promise` to ensure it is done before the callback completes.

What was the reasoning behind changing it like this?

Reply:
There seems to be more `async/await` syntax than `then/catch`, my only reason was consistency. Personally I prefer 2-3 extra lines but the same syntax everywhere, It just easier to read when you open multiple files or someone else's code.

Reply:
I definitely understand wanting consistency, but I feel like things aren't always as 'clean cut' as that. There are places (usually more complex code) where using `async`/`await` will make cleaner/easier to understand code. There are other places (such as this one), where using `.then()`/`.catch()` may be simpler. I feel like it's more about knowing when to use the 'right' pattern for the thing you need to solve, rather than making sure everything is exactly the same.

Basically the same concept as https://www.mentalfloss.com/article/625404/equity-vs-equality-what-is-the-difference (but in code pattern terms)